okhttp/src/main/java/okhttp3/internal/connection/StreamAllocation.java

```diff
@@ -298,6 +299,8 @@ public void streamFailed(IOException e) {
           noNewStreams = true;
           route = null;
         }
+      } else if (e instanceof ConnectionShutdownException) {
+        noNewStreams = true;
```

Comment:
I’m wondering if the case below which is special cased for multiplexed connections actually shouldn’t be special cased that way? Maybe instead it should be like this?

```
  noNewStreams = !(e instanceof StreamResetException);
```